lib/HotModuleReplacementPlugin.js

```diff
@@ -253,7 +253,12 @@ module.exports = class HotModuleReplacementPlugin {
 							c: {}
 						};
 						for (const key of Object.keys(records.chunkHashs)) {
-							const chunkId = isNaN(+key) ? key : +key;
+							let chunkId;
+							if (compiler.options.optimization.namedChunks) {
+								chunkId = key;
+							} else {
+								chunkId = isNaN(+key) ? key : +key;
+							}
 							const currentChunk = compilation.chunks.find(
 								chunk => chunk.id === chunkId
```

Comment:
``chunk => `${chunk.id}` === key``

Stringify chunk of instead of numberifing key

Reply:
Cool, get it. i'm trying add a test case at the PR